libs/langchain/langchain/callbacks/message_log.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import annotations
+
+import asyncio
+import math
+from typing import Any, AsyncIterator, Callable, Dict, Optional, TypedDict, Union
+from uuid import UUID
+
+import jsonpatch
+from anyio import create_memory_object_stream
+
+from langchain.callbacks.tracers.base_async import AsyncBaseTracer
+from langchain.callbacks.tracers.schemas import Run
+from langchain.schema.output import ChatGenerationChunk, GenerationChunk
+
+
+class Message(TypedDict):
+    name: str
+    run_type: str
+    tags: list[str]
+    metadata: dict[str, Any]
+
+    streamed_output: list[str]
+    final_output: Optional[Any]
+
+
+class MessageState(TypedDict):
+    messages: list[Message]
+    output: Optional[Any]
+
+
+class MessageLog:
+    state: Optional[MessageState]
+    """List of messages."""
+    patch_log: list[dict]
+    """List of jsonpatch operations that created the messages."""
+
+    def __init__(
+        self, patch_log: list[dict], state: Optional[MessageState] = None
+    ) -> None:
+        self.patch_log = patch_log
+        self.state = state
+
+    def __add__(self, other: Union[MessageLog, Any]) -> MessageLog:
```

Comment:
why not type as just MessageLog

Reply:
Because we dont control how people call this method

Reply:
isn't that true of any method?

but also Union[MessageLog, Any] == Any, so might as well type as that then?